src/common/ical.rs

```diff
@@ -0,0 +1,142 @@
+use icalendar::{Calendar, Component};
+
+use crate::handlers::error::ResourceError;
+
+pub async fn parse_ical_uri(
+    user_id: String,
+    ical_uri: String,
+) -> Result<Vec<crate::entity::event::EventEntity>, ResourceError> {
+    // Request source
+    let client = reqwest::Client::new();
+    let ical_body = client
+        .get(&ical_uri.clone())
+        .send()
+        .await
+        .map_err(|e| ResourceError::NetworkError)?
+        .text()
+        .await
+        .map_err(|e| ResourceError::FailedParse("Failed to read source response".to_string()))?;
```

Comment:
FailedParse adds the string `Could not parse` before your error message, so this error message will look pretty weird. Error messages should be as formatted as possible, that means that we add as less information as possible and have things like ResourceError::FailedParse for the formatting and text around it. This makes all errors more uniform and easily updateable.

Reply:
This error occurs when the reqwest library cannot decode and parse the full response body of the response. Would `ResourceError::FailedParse("response body".to_string())` be appropriate?